### ISSUE TITLE
Incorrect partial attribute (EAV) reindex (Update by Schedule) for configurable product with childs visibility "Not Visible Individually"

### ISSUE THREAD

**Issue by Nix-id** [magento/magento2#12667]
### Preconditions

1. Magento 2.2.1
2. PHP 7.1
3. MariaDB

### Steps to reproduce

0. Create config product and childs
For example with id '1',  and his childs with '2','3'
1. set indexer method to "Update by Schedule"
2. Edit and save only Configurable product (don't edit childs) 
3. waite while cron update MView
4. check DB `catalog_product_index_eav`. all entries with `entity_id`=1 and `source_id` in (2,3) are deleted. because of this, the filters on the catalog list page do not work correctly

### Expected result

1. All index must be reindex correct

### Actual result

1. 

### Reson

In file
https://github.com/magento/magento2/blob/2.2-develop/app/code/Magento/Catalog/Model/Indexer/Product/Eav/AbstractAction.php
method 
`processRelations($indexer, $ids, $onlyParents = false)`

if in $ids = only parent id  then
`$parentIds = $indexer->getRelationsByChild($ids);`
return empty result
and
`$childIds = $onlyParents ? [] : $indexer->getRelationsByParent($parentIds);`
also empty

### Fast solution
```
$parentIds = $indexer->getRelationsByChild($ids);

+$parentIds = array_unique(array_merge($parentIds, $ids));

$childIds = $onlyParents ? [] : $indexer->getRelationsByParent($parentIds);
```

**Comment by magento-engcom-team** [comment]
@Nix-id, thank you for your report.
We've created internal ticket(s) MAGETWO-85597 to track progress on the issue.

**Comment by Nix-id** [comment]
also i cant find any method how i can override method processRelations

redeclare abstract class - no
redeclare classes  Magento\Catalog\Model\Indexer\Product\Eav\Action\Full | Row | Rows
generate error:
PHP Fatal error:  Uncaught TypeError: Argument 3 passed to Magento\Catalog\Model\Indexer\Product\Eav::__construct() must be an instance of Magento\Catalog\Model\Indexer\Product\Eav\Action\Full, instance of Mag\IndexEAV\Fix\Full given, called in ../vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php on line 111 and defined in ../vendor/magento/module-catalog/Model/Indexer/Product/Eav.php:37

also i try crate plugin before*
and also no affect on result...

**Comment by okorshenko** [comment]
Hi @Nix-id 
The issue has been fixed and delivered in 2.2-develop branch. Will be available with upcoming patch release.
